### ISSUE TITLE
Error in loadSQM "cannot allocate vector of size ..."

### ISSUE THREAD

**Issue by fpusan** [jtamames/SqueezeMeta#386]
Hi,

I continued and finally got to load the files on R. I had several issues with the version of SQM tools, but was finally successful to install the latest one (only using the .tar.gz, not the .zip file). 

I then do this:

> library('SQMtools')
> NSt_ext = loadSQM('D:/T_Praga/Project_naph_st/Squeeze/NSt_ext')

Error: no se puede ubicar un vector de tamaño  62.5 Mb
Además: Warning message:
In sprintf("%s/results/tables/%s.superkingdom.%s.abund.tsv", project_path,  :
  one argument not used by format '%s/results/tables/%s.superkingdom.%s.abund.tsv'

Then if I try to load in SQMlite:

Error in loadSQMlite("D:/T_Praga/Project_naph_st/Squeeze/NSt_ext") : 
  Directory "D:/T_Praga/Project_naph_st/Squeeze/NSt_ext" does not seem to contain valid SqueezeMeta tables.

Not sure what I am doing wrong now. Otherwise I was trying to reproduce some of the charts in the SQM tutorial to analyze metagenomic data. Since I have generated the tsv files that were intended for SQM, all can be either parsed or used in other formats (although it takes more time than what I thought would be SQM).

_Originally posted by @jjgonzalezplaza in https://github.com/jtamames/SqueezeMeta/issues/377#issuecomment-968112412_

**Comment by fpusan** [comment]
Hi, @jjgonzalezplaza
It would seem that you ran out of memory? Is this your personal laptop? How large was the project?
For loading the project as a SQMlite object, you have to point to the tables directory (the output of `sqm2tables.py` or `sqmreads2tables.py`), not the project directory.
If you let SQMtools run that step for you, it will probably be
`loadSQMlite("D:/T_Praga/Project_naph_st/Squeeze/NSt_ext/results/tables") `

**Comment by jjgonzalezplaza** [comment]
Yes, I currently don't have access (physical) to the server where I have the data that has an Ubuntu system. I only access through Mobaxterm from my laptop with Windows 10. 

Regarding the SQMlite now is loading, but I get the following message

In sprintf("%s/%s.superkingdom.%s.abund.tsv", tables_path, project_name,  :
  one argument not used by format '%s/%s.superkingdom.%s.abund.tsv'

**Comment by jjgonzalezplaza** [comment]
OK, I must look like a complete newbie, but until not so long ago I was mainly a molecular biologist-microbiologist. I actually was able to launch Rstudio in the server (typing rstudio, surprisingly it did prompt an Rstudio version) (although it responds slightly slower than in my laptop). I'm testing now if I can load the whole project via loadSQM.
To be honest, being able to access Rstudio at my server is mind-blowing.

Question now is, shouldn't be a way to launch this session directly on my laptop RStudio application?

**Comment by jjgonzalezplaza** [comment]
Update: managed to load SQMlite, but it does not compute anything for the ORF table (gives back a NULL value). I managed to get the taxonomy plot and also the functions plot (KO). 

In the server Rstudio version I get the following:
Error in file(file, "rt") : cannot open the connection
In addition: Warning message:
In file(file, "rt") :
  cannot open file '/media/termite/TRT_Server2/SEQS/P_NSt_met/NSt_ext/results/tables/NSt_ext.contig.tax.tsv': No such file or directory

I tried to run again the sqm2tables.py NSt_ext NSt_ext/results/tables --force-overwrite, but still no success geting the missing table. 
The version of sqmtools is 0.6.2

**Comment by fpusan** [comment]
Hi!
Regarding your first question, a SQMlite object does not have ORF information. You can read more details in the documentation ofr the R package.
Regarding the second one, it is a bit weird. Maybe you are not using the latest version of `sqm2tables.py`? I would test that before trying to cheat ;)

**Comment by jjgonzalezplaza** [comment]
Hi,

I did again the step 09 by using the script for that step. Then again running sqm2tables.py deleting the previous folder. I checked that the version of sqm2tables.py was the correct one. 
In R I did load again data.table package by downloading the tar.gz from: https://cran.r-project.org/web/packages/data.table/index.html.
I think that this was one of the problems. Then re-install again SQMtools in R. Re-start R and load my project as:
loadSQM('/media/termite/TRT_Server2/SEQS/P_NSt_met/NSt_ext/', tax_mode = "allfilter", trusted_functions_only = F, engine = "data.frame"). 
When not using the latest version, the option "engine" does not appear. That's why I realized that my version of SQMtools was not correct.

All solved now, it does work without issues. I have to check still how the Krona export works, but if it fails I will open a new issue. 

Thanks! I can finally write the results and discussion sections.
